### ISSUE TITLE
Make cabal a first class citizen for CI

### ISSUE THREAD

**Issue by fendor** [haskell/haskell-ide-engine#1431]
CI hardly uses cabal, although it is one of the two supported installation methods.
For all supported GHC Versions, we should add a CI option to build HIE with cabal.
Also, we should run the tests with cabal as well.

**Comment by jneira** [comment]
I am adding cabal jobs with tests for ghc-8.6.5 and ghc-8.4.4: https://github.com/haskell/haskell-ide-engine/pull/1571
@fendor It would be enough to close this or you think all supported ghcs will be needed?
or maybe it should be added to circle ci...

**Comment by fendor** [comment]
Good enough for me! Tests have still a run-time dependency on stack, right?

**Comment by jneira** [comment]
Yeah, the `CabalHelper` unit tests fails if stack (or cabal) is not in path
